### DISCUSSION TITLE
Welcome to Nuxt File Storage Discussions!

### DISCUSSION THREAD

**Discussion by NyllRE** [NyllRE/nuxt-nitro-storage#9] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by nowo** [comment]
I have an idea to add an option 'maxStreamSize'
```ts
// nuxt.config.ts
fileStorage: {
  ...
  maxStreamSize: 100
}
```
For example, if the file size is greater than 100MB, use `createWriteStream`, default use `writeFile`

**Reply by NyllRE** [reply]
hey @nowo! Interesting idea, though now that I think about it wouldn't this generally be the best option in general? If this use case exists then what would be the reason to use the base64 method? also I wonder how the backend of accepting this file would look like

**Reply by nowo** [reply]
> hey @nowo! Interesting idea, though now that I think about it wouldn't this generally be the best option in general? If this use case exists then what would be the reason to use the base64 method? also I wonder how the backend of accepting this file would look like

'createWriteStream' for large file write save, performance will improve, it looks like this

```ts
import fs from 'node:fs'

const saveStreamFile = (filePath: string, fileData: Buffer) => {
    return new Promise((resolve, reject) => {

        if (!fs.existsSync(filePath)) {
            fs.mkdirSync(filePath);
        }

        const stream = fs.createWriteStream(filePath);

        stream.on('open', () => {
            const blockSize = 128;
            const nbBlocks = Math.ceil(fileData.length / (blockSize));
            for (let i = 0; i < nbBlocks; i += 1) {
                const currentBlock = fileData.slice(
                    blockSize * i,
                    Math.min(blockSize * (i + 1), fileData.length),
                );
                stream.write(currentBlock);
            }

            stream.end();
        });
        stream.on('error', (err) => { reject(err); });
        stream.on('finish', () => { resolve(true); });
    });
}


// data is Buffer
if (data.length > useRuntimeConfig().fileStorage.maxStreamSize) {
    await saveStreamFile('/upload/file.zip', data)
    ...
} else {
    ...
}

```
